index.js

```diff
@@ -26,17 +26,26 @@ function getResources(dir, config) {
 /**
  * Smokes the given resources.
  */
-function smoke(resources, currentTry, config) {
+function smoke(resources, currentTry, config, callback) {
   shisha.smoke(resources, function(report){
-    var failedResources = _.filter(report, function(resource){
+    var failedResources = _(report).filter(function(resource){
       return !(resource.result);
-    })
+    }).map(function(item) {
+      item.url = item.url();
+      item.status = 200;
+      return item;
+    }).value();
 
     if (Object.keys(failedResources).length) {
       setTimeout(function(){
-        retry(failedResources, currentTry, config);
+        try {
+          retry(failedResources, currentTry, config, callback);
+        } catch (err) {
+          callback(err);
+        }
```

Comment:
do we need this try catch considering that the callback just rethrows the exception?